pylops/medical/ct.py

```diff
@@ -86,54 +92,125 @@ class CT2D(LinearOperator):
+                self.projector_type = "cuda"
+            else:
+                raise ValueError(
+                    "Only 'cuda' projector type is supported for 'cuda' engine."
+                )
+
+        # create create volume and projection geometries as well as projector
+        self._init_geometries()
+        if engine == "cuda":
+            # efficient GPU data exchange only implemented for 3D data in ASTRA, so we
+            # emulate 2D geometry as 3D case with 1 slice
+            self._init_1_slice_3d_geometries()
+
+        super().__init__(
+            dtype=np.dtype(dtype), dims=dims, dimsd=(len(thetas), det_count), name=name
+        )
+
+    def _init_geometries(self):
+        self.vol_geom = astra.create_vol_geom(self.dims)
+        if self.proj_geom_type == "parallel":
             self.proj_geom = astra.create_proj_geom(
-                proj_geom_type, det_width, det_count, thetas
+                "parallel", self.det_width, self.det_count, self.thetas
             )
-        else:
+        elif self.proj_geom_type == "fanflat":
             self.proj_geom = astra.create_proj_geom(
-                proj_geom_type,
-                det_width,
-                det_count,
-                thetas,
-                source_origin_dist,
-                origin_detector_dist,
+                "fanflat",
+                self.det_width,
+                self.det_count,
+                self.thetas,
+                self.source_origin_dist,
+                self.origin_detector_dist,
             )
-
-        # create projector
-        self.proj_id = astra.create_projector(
-            projector_type, self.proj_geom, self.vol_geom
+        self.projector_id = astra.create_projector(
+            self.projector_type, self.proj_geom, self.vol_geom
         )
-        super().__init__(
-            dtype=np.dtype(dtype), dims=dims, dimsd=(len(thetas), det_count), name=name
+
+    def _init_1_slice_3d_geometries(self):
+        self._3d_vol_geom = astra.create_vol_geom(*self.dims, 1)
+        if self.proj_geom_type == "parallel":
+            self._3d_proj_geom = astra.create_proj_geom(
+                "parallel3d", 1.0, self.det_width, 1, self.det_count, self.thetas
+            )
+        elif self.proj_geom_type == "fanflat":
+            self._3d_proj_geom = astra.create_proj_geom(
+                "cone",
+                1.0,
+                self.det_width,
+                1,
+                self.det_count,
+                self.thetas,
+                self.source_origin_dist,
+                self.origin_detector_dist,
+            )
+        self._3d_projector_id = astra.create_projector(
+            "cuda3d", self._3d_proj_geom, self._3d_vol_geom
         )
 
     @reshaped
     def _matvec(self, x):
-        y_id, y = astra.create_sino(x, self.proj_id)
-        astra.data2d.delete(y_id)
+        ncp = get_array_module(x)
+        backend = get_module_name(ncp)
+        if backend == "numpy":
```

Comment:
Ok maybe I see here what is your second case... if you have engine="cuda" but pass numpy you will go for the else branch and I guess internally in `astra.experimental.direct_FP3D` x is moved to the GPU?

If this is the case maybe we can leave the 3 cases as you had and let the user to choose if they want to lift data to the GPU (especially if they want it to flow between operators in a chain) or let ASTRA do it if they want numpy in - numpy out...